src/ArrowEffects.cpp

```diff
@@ -337,9 +343,9 @@ void ArrowEffects::Update()
 
 		if( !position.m_bFreeze || !position.m_bDelay )
 		{
-			data.m_fExpandSeconds += fTime - fLastTime;
+			data.m_fExpandSeconds += static_cast<float>(fTime - fLastTime);
```

Comment:
Don't we lost precision here?

Reply:
`fTime` and `fLastTime` are declared as `double`, so the subtraction is performed using `double` precision. The `PerPlayerStruct` stores `float`s,  so only the final result is cast to `float`.